tensorflow/core/kernels/training_ops.cc

```diff
@@ -328,6 +328,45 @@ struct ApplyAdamSYCL {
 template <typename T>
 struct ApplyAdam<CPUDevice, T> : ApplyAdamNonCuda<CPUDevice, T> {};
 
+template <typename Device, typename T>
+struct ApplyAdaMaxNonCuda {
+  void operator()(const Device& d, typename TTypes<T>::Flat var,
+                  typename TTypes<T>::Flat m, typename TTypes<T>::Flat v,
+                  typename TTypes<T>::ConstScalar beta1_power,
+                  typename TTypes<T>::ConstScalar beta2_power,
+                  typename TTypes<T>::ConstScalar lr,
+                  typename TTypes<T>::ConstScalar beta1,
+                  typename TTypes<T>::ConstScalar beta2,
+                  typename TTypes<T>::ConstScalar epsilon,
+                  typename TTypes<T>::ConstFlat grad, bool use_nesterov) {
+    if (use_nesterov) {
+      LOG(WARNING) << "AdaMax doesn't support use_nesterov yet, ignore it.";
+    }
+    m.device(d) += (grad - m) * (T(1) - beta1());
+    // v == u
+    v.device(d) = (beta2() * v).cwiseMax(grad.abs());
+    // var == θ
+    var.device(d) -= lr() / (T(1) - beta1_power()) * (m / (v + epsilon()));
+  }
+};
+
+#ifdef TENSORFLOW_USE_SYCL
```

Comment:
Is this tested? If not, I'd rather not have the code right now..

Reply:
I didn't test the code because I have not any machine with SYCL environment.

Reply:
Yes

Reply:
Could you give me an introduction (or example) how to add SYCL test?